numpy/_core/src/multiarray/multiarraymodule.c

```diff
@@ -3274,161 +3275,163 @@ PyArray_Where(PyObject *condition, PyObject *x, PyObject *y)
     NPY_cast_info cast_info = {.func = NULL};
 
     ax = (PyArrayObject*)PyArray_FROM_O(x);
+    if (ax == NULL) {
+        goto fail;
+    }
     ay = (PyArrayObject*)PyArray_FROM_O(y);
-    if (ax == NULL || ay == NULL) {
+    if (ay == NULL) {
         goto fail;
```

Comment:
Sorry for writing this with awkward control flow in the first place... :upside_down_face: